server/routes/calendar.js

```diff
@@ -112,4 +113,65 @@ router.get("/availability/:userId", (req, res) => {
   );
 });
 
+const coverColors = ["#ffff42", "#ffaf0f", "#ffc34d", "#4eb548"];
+
+router.post("/openshifts/:userId", (req, res) => {
+  let shifts = req.body.items;
+  pool.query(
+    "select * from coverrequests full join shifts on coverrequests.shift_id = shifts.shift_id where coverer_id is not distinct from $1 and request_id is distinct from $1",
```

Comment:
This query isn't ignoring coverrequests put up by the current user. Also it can be simplified a little:
- change full join to inner join
- that allows you to delete the "and request_id is distinct from $1" part
- change the first where condition to: coverer_id is null and then get rid of the null in the parameter list